analyzers/tests/SonarAnalyzer.UnitTest/Rules/DebuggerDisplayUsesExistingMembersTest.cs

```diff
@@ -0,0 +1,67 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using CS = SonarAnalyzer.Rules.CSharp;
+using VB = SonarAnalyzer.Rules.VisualBasic;
+
+namespace SonarAnalyzer.UnitTest.Rules;
+
+[TestClass]
+public class DebuggerDisplayUsesExistingMembersTest
+{
+    private readonly VerifierBuilder builderCS = new VerifierBuilder<CS.DebuggerDisplayUsesExistingMembers>();
+    private readonly VerifierBuilder builderVB = new VerifierBuilder<VB.DebuggerDisplayUsesExistingMembers>();
+
+    [TestMethod]
+    public void DebuggerDisplayUsesExistingMembers_CS() =>
+        builderCS.AddPaths("DebuggerDisplayUsesExistingMembers.cs").Verify();
+
+#if NET
```

Comment:
Not all new language features require .NET Core / .NET. I think the `DebuggerDisplayUsesExistingMembers_CSharp8` and `DebuggerDisplayUsesExistingMembers_CSharp11` can run on .NET Framework as well.
`DebuggerDisplayUsesExistingMembers_CSharp9` and `DebuggerDisplayUsesExistingMembers_CSharp10` unfortunately won't. They should be (`record` types are just syntactic sugar, they are supported by the .NET Framework), but unfortunately there's bug in .NET 5 that doesn't allow it.
Regardless, I'd move out `DebuggerDisplayUsesExistingMembers_CSharp8` and `DebuggerDisplayUsesExistingMembers_CSharp11` from the `#if` block, just to get better coverage.

Reply:
It seems that everything here works for both .NET Core and .NET Framework.

I have removed the `#if`.